### PR TITLE
doc: release-notes-4.0: Add ethernet/mdio notes

### PR DESCRIPTION
**Pull request by decsny**
Add release notes for ethernet and mdio drivers for 4.0

### CODE REVIEW

**Comment by pdgendt** [comment]
Check compliance issues

**Review by pdgendt — APPROVED** [review]

**Comment by jukkar** [comment]
I also added Ethernet stuff in #81216 but your changes better.

**Comment by jukkar** [comment]
@decsny I am fine if you create the Ethernet driver release notes now and in the future. I just did it from old habit as there is no maintainer atm for Ethernet drivers. Perhaps you would like to volunteer to be a maintainer for the Ethernet drivers?

**Review by jukkar — APPROVED** [review]

**Comment by decsny** [comment]
@jukkar sorry about this duplication, I didn't know someone else will do it. What is the difference between maintainer and collaborator ?

**Comment by jukkar** [comment]
> sorry about this duplication, I didn't know someone else will do it.

No worries at all, I did not knew either that someone was writing the Ethernet driver part. Anyway, your PR looks good so we can use it for the notes.

> What is the difference between maintainer and collaborator ?

See https://docs.zephyrproject.org/latest/project/project_roles.html

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -369,9 +369,67 @@ Drivers and Sensors
 
 * Ethernet
 
-  * LiteX: Renamed the ``compatible`` from ``litex,eth0`` to :dtcompatible:`litex,liteeth`.
-  * STM32: Driver can now be configured to use a preemptive RX thread priority, which could be useful
-    in case of high network traffic load (reduces jitter).
```
Comment (mmahadevan108) [inline doc/releases/release-notes-4.0.rst:374]: Why was this portion deleted? Or is this how github is displaying the change?

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**